### PR TITLE
Initial setup: ShopifyOrderProcessingFunction and CreateOrderFunction

### PR DESCRIPTION
**Pull request by infinity-cloud-solutions**
- Lambdas ShopifyOrderProcessingFunction and CreateOrderFunction
- Basic happy path test ShopifyOrderProcessingFunction

### CODE REVIEW

File: sam-hiberry-app/template.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    Timeout: 3
+
+Parameters:
+  ShopifyEventBusName:
+    Type: String
+  StageName:
+    Type: String
+    Default: uat
+    AllowedValues:
+      - uat
```
Comment (MarcoBurgos) [inline sam-hiberry-app/template.yaml:19]: should we have also a dev env

File: sam-hiberry-app/tests/events/shopify_event.json [hunk 2]
```diff
@@ -0,0 +1,438 @@
+{
+    "version": "0",
+    "id": "a575d0e8-2e37-0b37-b1f5-29e8be49566f",
+    "detail-type": "shopifyWebhook",
+    "source": "aws.partner/shopify.com/71471267841/ShopifySourceTest",
```
Comment (MarcoBurgos) [inline sam-hiberry-app/tests/events/shopify_event.json]: if this is for test, we could anonymize some of the values

File: sam-hiberry-app/create_order/app.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            "created_at": data.get("created_at", datetime.now().isoformat()),
+            "updated_by": int(data.get("updated_by", 0)),
+            "updated_at": data.get("updated_at", datetime.now().isoformat()),
+            #"errors": data.get("errors", []), 
+            #"notes": data.get("notes"),
+            #"status": data.get("status"),
+            #"delivery_sequence": int(data.get("delivery_sequence", 0))
+        }
+        
+        print(f'Order received and created: {order}')
```
Comment (MarcoBurgos) [inline sam-hiberry-app/create_order/app.py]: should we use the logger, instead?

File: sam-hiberry-app/create_order/app.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        return {
+            'statusCode': 200,
```
Comment (MarcoBurgos) [inline sam-hiberry-app/create_order/app.py]: should be 201? https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201

File: sam-hiberry-app/create_order/app.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            'body': json.dumps('Order received and created successfully.')
+        }
+
+    except Exception as e:
```
Comment (MarcoBurgos) [inline sam-hiberry-app/create_order/app.py]: i would suggest adding more exceptions, for validations or for unauthorized access, so we can return a 404, 400 or 403

File: sam-hiberry-app/order_processing/app.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+# Initialize logging
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+# Shopify identifier
+SHOPIFY_ID = 0
+
+CREATE_ORDER_API_URL = os.getenv('CREATE_ORDER_API_URL', None)
+
+def process_shopify_event(shopify_data: dict) -> dict:
```
Comment (MarcoBurgos) [inline sam-hiberry-app/order_processing/app.py]: i would suggest, create another module for this one

File: sam-hiberry-app/order_processing/app.py [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+import requests
+
+# Initialize logging
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+# Shopify identifier
+SHOPIFY_ID = 0
+
+CREATE_ORDER_API_URL = os.getenv('CREATE_ORDER_API_URL', None)
```
Comment (MarcoBurgos) [inline sam-hiberry-app/order_processing/app.py]: if this is not set, it will fail

File: sam-hiberry-app/template.yaml (hunk 1 above)
Comment (infinity-cloud-solutions) [reply sam-hiberry-app/template.yaml:19]: added local and dev

File: sam-hiberry-app/tests/events/shopify_event.json (hunk 2 above)
Comment (infinity-cloud-solutions) [reply sam-hiberry-app/tests/events/shopify_event.json]: fields anonymized

File: sam-hiberry-app/create_order/app.py (hunk 3 above)
Comment (infinity-cloud-solutions) [reply sam-hiberry-app/create_order/app.py]: using powertools logger instead

File: sam-hiberry-app/order_processing/app.py (hunk 6 above)
Comment (infinity-cloud-solutions) [reply sam-hiberry-app/order_processing/app.py]: applied

File: sam-hiberry-app/order_processing/app.py (hunk 7 above)
Comment (infinity-cloud-solutions) [reply sam-hiberry-app/order_processing/app.py]: Raising ConfigurationError instead . In theory it shouldn't fail on AWS, as it is injected by sam.

**Review by MarcoBurgos — APPROVED** [review]

**Pull request merged**